sdk/src/trace/span.cc

```diff
@@ -98,6 +101,41 @@ Span::Span(std::shared_ptr<Tracer> &&tracer,
     return true;
   });
 
+  links.ForEachKeyValue([&](trace_api::SpanContext span_context, nostd::string_view key,
+                            opentelemetry::common::AttributeValue value, bool is_last_kv) noexcept {
```

Comment:
If we could the callback to accept a `KeyValueIterable`, then we could drastically simplify this code.